src/ansys/mechanical/core/run.py

```diff
@@ -110,6 +110,10 @@ def _cli_impl(
         if input_script:
             raise Exception("Cannot open in server mode with an input script.")
 
+    # If the input_script and port are missing in batch mode, raise an exception
+    if (not graphical) and (input_script is None) and (not port):
+        raise Exception("An input script, -i, or port, --port, are required in batch mode.")
+
```

Comment:
Why not just put them as required arguments via click? Click will take care of this for you... no need to do the if.

See https://click.palletsprojects.com/en/8.1.x/options/#basic-value-options

```py
@click.option(..., required=True,...)
```